[Bug]: Failed to manually update netdata on macOS from v1.33.1-28-nightly

### Bug description

I'm trying to manually update netdata agent from v1.33.1-28-nightly to latest but re-installation fails:

```
mac-3:netdata` copi$ curl https://my-netdata.io/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh --claim-token TOKEN --claim-rooms ROOM --claim-url https://app.netdata.cloud
...
CXX      aclk/aclk-schemas/proto/chart/v1/stream.pb.o
collectors/macos.plugin/macos_fw.c:86:18: warning: implicit declaration of function 'IOMainPort' is invalid in C99 [-Wimplicit-function-declaration]
    if (unlikely(IOMainPort(bootstrap_port, &main_port))) {
                 ^
1 warning generated.
  CXX      aclk/aclk-schemas/proto/chart/v1/instance.pb.o
  CXX      aclk/aclk-schemas/proto/chart/v1/dimension.pb.o
  CXX      aclk/aclk-schemas/proto/chart/v1/config.pb.o
  CXX      aclk/aclk-schemas/proto/aclk/v1/lib.pb.o
  CXX      aclk/aclk-schemas/proto/alarm/v1/config.pb.o
  CXX      aclk/aclk-schemas/proto/alarm/v1/stream.pb.o
  CXX      aclk/aclk-schemas/proto/nodeinstance/info/v1/info.pb.o
  CC       cli/cli.o
  CCLD     cups.plugin
  GEN      netdatacli
  GEN      netdata
Undefined symbols for architecture x86_64:
  "_IOMainPort", referenced from:
      _do_macos_iokit in macos_fw.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [netdata] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
 FAILED   

Wed Mar 30 12:07:43 EEST 2022 : ERROR:  FAILED TO UPDATE NETDATA : FAILED TO COMPILE/INSTALL NETDATA
 FAILED 

 ABORTED  Failed to update existing Netdata install at /usr/local/netdata
```

Full log: https://pastebin.com/AzEHYvjL


### Expected behavior

The update should complete successfully.

### Steps to reproduce

- On a macOS 12.1 try to update/reinstall netdata from v1.33.1-28-nightly using kickstart


### Installation method

kickstart.sh

### System info

```shell
Darwin mac-3 21.2.0 Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:54 PST 2021; root:xnu-8019.61.5~1/RELEASE_X86_64 x86_64
```


### Netdata build info

```shell
Version: netdata v1.33.1-28-nightly
Configure options:  '--prefix=/usr/local/netdata/usr' '--sysconfdir=/usr/local/netdata/etc' '--localstatedir=/usr/local/netdata/var' '--libexecdir=/usr/local/netdata/usr/libexec' '--libdir=/usr/local/netdata/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--enable-cloud' '--with-bundled-protobuf' '--with-bundled-libJudy' 'CFLAGS= -I/usr/local/opt/openssl@1.1/include -I/usr/local/include' 'LDFLAGS= -L/usr/local/opt/openssl@1.1/lib -L/usr/local/lib'
Install type: kickstart-build
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES 
    ACLK Next Generation:       YES
    ACLK-NG New Cloud Protocol: YES
    ACLK Legacy:                NO
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         YES
Libraries:
    protobuf:                YES (bundled)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    NO
    cgroup Network Tracking: NO
    CUPS:                    YES
    EBPF:                    NO
    IPMI:                    NO
    NFACCT:                  NO
    perf:                    NO
    slabinfo:                NO
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES
```


### Additional info

_No response_

--- a/collectors/macos.plugin/macos_fw.c
+++ b/collectors/macos.plugin/macos_fw.c
@@ -78,7 +78,7 @@ int do_macos_iokit(int update_every, usec_t dt) {
     // NEEDED BY: do_bandwidth
     struct ifaddrs *ifa, *ifap;
 
-#if !__is_identifier(IOMainPort) /* macOS >= 12.0 */
+#if !defined(MAC_OS_VERSION_12_0) || (MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_VERSION_12_0)
 #define IOMainPort IOMasterPort
 #endif
 

